include/deal.II/fe/mapping.h

```diff
@@ -823,6 +723,156 @@ class Mapping : public Subscriptor
                           dealii::internal::FEValues::MappingRelatedData<dim, spacedim> &output_data) const = 0;
 
   /**
+   * @}
+   */
+
+public:
+
+  /**
+   * @name Functions usable by finite element fields to map points and tensors
+   * @{
+   */
+
+  /**
+   * Transform a field of vectors or 1-differential forms according to the
+   * selected MappingType.
+   *
+   * @note Normally, this function is called by a finite element, filling
+   * FEValues objects. For this finite element, there should be an alias
+   * MappingType like @p mapping_bdm, @p mapping_nedelec, etc. This alias
+   * should be preferred to using the types below.
+   *
+   * The mapping types currently implemented by derived classes are:
+   * <ul>
+   * <li> @p mapping_contravariant: maps a vector field on the reference cell
+   * is to the physical cell through the Jacobian:
+   * @f[
+   * \mathbf u(\mathbf x) = J(\mathbf{\hat x})\mathbf{\hat u}(\mathbf{\hat x}).
+   * @f]
+   * In physics, this is usually referred to as the contravariant
+   * transformation. Mathematically, it is the push forward of a vector field.
+   *
+   * <li> @p mapping_covariant: maps a field of one-forms on the reference
+   * cell to a field of one-forms on the physical cell. (theoretically this
+   * would refer to a DerivativeForm<1, dim, 1> but it canonically identified
+   * with a Tensor<1,dim>). Mathematically, it is the pull back of the
+   * differential form
+   * @f[
+   * \mathbf u(\mathbf x) = J(J^{T} J)^{-1}(\mathbf{\hat x})\mathbf{\hat
+   * u}(\mathbf{\hat x}).
+   * @f]
+   * In the case when dim=spacedim the previous formula reduces to
+   * @f[
+   * \mathbf u(\mathbf x) = J^{-T}(\mathbf{\hat x})\mathbf{\hat
+   * u}(\mathbf{\hat x}).
+   * @f]
+   * Gradients of scalar differentiable functions are transformed this way.
+   *
+   * <li> @p mapping_piola: A field of <i>n-1</i>-forms on the reference cell
+   * is also represented by a vector field, but again transforms differently,
+   * namely by the Piola transform
+   * @f[
+   *  \mathbf u(\mathbf x) = \frac{1}{\text{det}J(\mathbf x)}
+   * J(\mathbf x) \mathbf{\hat u}(\mathbf x).
+   * @f]
+   * </ul>
+   *
+   * @todo What is n in mapping_piola description?
```

Comment:
Can you remove the ```@todo``` and add a line that the ```n``` stands for the space dimension? (Applies also to other places.)

The formulas definitely deserve some checking. But I'm to tired right now to read pure latex math statements. :-]